Network/DNS/Types.hs

```diff
@@ -405,9 +411,66 @@ data DNSFlags = DNSFlags {
                            -- available in the name server.
 
   , rcode        :: RCODE  -- ^ Response code.
-  , authenData   :: Bool   -- ^ Authentic Data (RFC4035).
+  , authenData   :: Bool   -- ^ AD bit (RFC4035, Section 3.2.3).
+  , chkDisable   :: Bool   -- ^ CD bit (RFC4035, Section 3.2.2).
   } deriving (Eq, Show)
 
+-- | 'DNSFlags' record that can hold only query-related values.
+-- The constructors is not exported, but 'mkQueryFlags' can
+-- create a valid record, which can be converted to generic
+-- 'DNSFlags' via 'fromQueryFlags'.
+--
+newtype QueryFlags = QueryFlags
+    { -- | 'DNSFlags' accessor.
+      fromQueryFlags :: DNSFlags }
+
+-- | Boolean type representing the Recursion Desired (RD) DNS header flag bit.
+-- See
+-- <https://tools.ietf.org/html/rfc1035#section-4.1.1 Header section format>
+-- and/or
+-- <https://tools.ietf.org/html/rfc2929#section-2 DNS Query/Response Headers>
+--
+newtype RDbit = RDbit { fromRDbit :: Bool }
+
+-- | Boolean type representing the Authenticated Data (AD) DNS header flag bit.
+-- See <https://tools.ietf.org/html/rfc4035#section-3.2.3 RFC4035 The AD bit>.
+-- The AD bit is only meaningful in recursive queries.
+--
+newtype ADbit = ADbit { fromADbit :: Bool }
+
+-- | Boolean type representing the Checking Disabled (CD) DNS header flag bit.
+-- See <https://tools.ietf.org/html/rfc4035#section-3.2.2 RFC4035 The CD bit>.
+-- The CD bit is only meaningful in recursive queries.
+--
+newtype CDbit = CDbit { fromCDbit :: Bool }
+
+-- | Return 'DNSFlags' appropriate for a query with the specified values of
+-- the RD, AD and CD bits.  This can be used to update the return value of
+-- 'defaultQuery' with the desired settings of those bits:
+--
+-- >>> :{
+-- let hdr = header defaultQuery
+--     qry = defaultQuery {
+--     header = hdr {
+--              flags = fromQueryFlags $
+--                      mkQueryFlags (RDbit True) (ADbit True) (CDbit False)
```

Comment:
I think that `defaultQueryFlags` should be defined somewhere and used here: `fromQueryFlags defaultQueryFlags`.